docs/core/context.md

```diff
@@ -4,7 +4,7 @@ order: 3
 
 # Context
 
-The `context` is a data structure intended to be passed from function to function that carries information about the current state of the application. It holds a cached copy of the entire state as well as useful objects and information like `gasMeter`, `block height`, `consensus parameters` and more. {synopsis}
+The `context` is a data structure intended to be passed from function to function that carries information about the current state of the application. It provides an access to a virtual storage (a safe branch of the entire state) as well as useful objects and information like `gasMeter`, `block height`, `consensus parameters` and more. {synopsis}
```

Comment:
Agree that "cache" might not be the correct verbiage, but I don't see how "virtual" is any better tbh. Both a parent `Context` and it's (cached - for lack of a better term) child `Context` both contain logical stores. What does "virtual" mean here exactly?

Reply:
Thanks for comment. My motivation for using  _virtual_ is by adding a new logical layer, similar to creating virtual resources in OS.  If this is misleading let's find better one. Few other proposals:
* store branch (i'm already using _branch_ so it can make sense)
* snapshot (Ethereum uses this term)
* version (simple and makes sense as well).

Reply:
I like snapshot! Branch sounds good too. Those are my order of preferences :)

Reply:
Snapshot sounds great

Reply:
When going through the docs, I've decided to use _branching_. Here is my motivation:
* state is an abstract term: a storage (as a whole) version at particular time.
* state has versions: we can create new versions and commit them to the _main line_ through merging.
* similarly to how it's in version control systems (eg git), to start a new version we create a branch. One branch can branch in another branch.

I wanted to use only one term, and _branch_ seams the most natural in this context.